### PR TITLE
Fix some skipped SignalR tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
* +WebSockets user agent!

### CODE REVIEW

File: src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs [hunk 1]
```diff
@@ -36,8 +36,13 @@ public WebSocketsTransport(HttpConnectionOptions httpConnectionOptions, ILoggerF
         {
             _webSocket = new ClientWebSocket();
 
-            // Issue in ClientWebSocket prevents user-agent being set - https://github.com/dotnet/corefx/issues/26627
-            //_webSocket.Options.SetRequestHeader("User-Agent", Constants.UserAgentHeader.ToString());
+            // Full Framework will throw when trying to set this
+            // Catching is easier and safer than trying to use #if !NETSTANDARD2_0
+            try
```
Comment (anurse) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs]: Let's do conditional complication and only call this in `#if NETCOREAPPX_Y`

**Comment by BrennanConroy** [comment]
Updated

File: src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs [hunk 2]
```diff
@@ -36,8 +36,11 @@ public WebSocketsTransport(HttpConnectionOptions httpConnectionOptions, ILoggerF
         {
             _webSocket = new ClientWebSocket();
 
-            // Issue in ClientWebSocket prevents user-agent being set - https://github.com/dotnet/corefx/issues/26627
-            //_webSocket.Options.SetRequestHeader("User-Agent", Constants.UserAgentHeader.ToString());
+            // Full Framework will throw when trying to set this
+            // So avoid setting it in netstandard2.0 and only set it in netstandard2.1 and higher
+#if !NETSTANDARD2_0
+            _webSocket.Options.SetRequestHeader("User-Agent", Constants.UserAgentHeader.ToString());
```
Comment (halter73) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs]: On full framework, could we set "X-SignalR-User-Agent" like we do in our TypeScript client?

File: src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs [hunk 3]
```diff
@@ -3289,9 +3289,9 @@ public async Task ConnectionAbortedIfSendFailsWithProtocolError()
                 {
                     var connectionHandlerTask = await client.ConnectAsync(connectionHandler).OrTimeout();
 
-                    await client.SendInvocationAsync(nameof(MethodHub.ProtocolError)).OrTimeout();
-
                     await client.Connected.OrTimeout();
```
Comment (halter73) [inline src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs]: Why do you need to wait on the ConnectedTask in this test?

File: src/SignalR/server/SignalR/test/WebSocketsTransportTests.cs [hunk 4]
```diff
@@ -86,7 +86,14 @@ public async Task WebSocketsTransportSendsUserAgent()
                     .Assembly
                     .GetCustomAttribute<AssemblyInformationalVersionAttribute>();
 
-                Assert.Equal("Microsoft.AspNetCore.Http.Connections.Client/" + assemblyVersion.InformationalVersion, userAgent);
+                Assert.StartsWith("Microsoft SignalR/", userAgent);
```
Comment (halter73) [inline src/SignalR/server/SignalR/test/WebSocketsTransportTests.cs]: I think it's OK to be stricter with this test. If we change the User-Agent format, that could break customers.

```suggestion
                Assert.StartsWith($"Microsoft SignalR/{majorVersion}.{minorVersion} ({assemblyVersion.InformationalVersion}; ", userAgent);
```

**Review by halter73 — APPROVED** [review]

**Pull request merged**